Instructions Page

o-bm opened the pull request:
Created the instructions page. It fetches instructions data (In JSON format) from the back end and render it dynamically to display to the user. 
It is very easy to add or modify the instructions as you can just make changes to the structure in InstructionsController.js. Each instruction has a title and a body that is used to display on frontend. 

Note that there is a previous branch called instructions_page but something got messed up because of an update to main that I needed  so I will delete it. Nothing to worry about there. 

Also I suggest that we maybe use the instructions page as a FAQ. I think this will be better for the application and we can address common issues or questions that users may have while still including the instructions on how to use the app.

close #32

arthurhohinng commented on client/src/pages/InstructionsPage.tsx:
```diff
@@ -0,0 +1,42 @@
+  body: string;
+}
+
+const InstructionsPage: React.FC = () => {
```
React.FC isn't preferred
https://github.com/facebook/create-react-app/pull/8177

arthurhohinng reviewed (commented):
Could I get an example of when the instructions page is used?
Also an image would be helpful

Vladmidir commented on client/src/pages/InstructionsPage.tsx:
```diff
@@ -0,0 +1,42 @@
+
+  useEffect(() => {
+    // Fetch the instructions from the backend to display
+    fetch('/api/instructions')
```
we can hardcode instructions on the client side.

Vladmidir reviewed (changes requested):
I strongly suggest you remove the server components.

Also, the ticket implied that you will write the actual instructions content.

o-bm commented on server/routes/InstructionsRoutes.js:
Agreed. Will have the instructions hardcoded and make it a react component

o-bm commented on server/index.js:
```diff
@@ -28,6 +28,8 @@ const challenges = require('./routes/ChallengeRoutes');
 const users = require('./routes/UserRoutes');
 const comments = require('./routes/CommentRoutes');
+const instructionsRoutes = require('./routes/InstructionsRoutes');
+app.use('/api', instructionsRoutes);
```
Thanks for pointing that out

o-bm commented on client/src/pages/InstructionsPage.tsx:
```diff
@@ -0,0 +1,42 @@
+import { Container, Accordion, Card } from 'react-bootstrap';
+
+// Define the structure of each instruction
+interface Instruction {
```
noted

o-bm commented on server/controllers/InstructionsController.js:
```diff
@@ -0,0 +1,10 @@
+const instructions = [
+    {title: "Getting Started", body: "This is a new instruction"},
```
will do

o-bm commented:
Removed backend integration for  instructions 
Made react component for instructions
Updated challenge page so instructions pop up when a user clicks on a challenge 

<img width="1270" alt="image" src="https://github.com/utmgdsc/UML_Mentor/assets/84227399/564601f5-1f8a-4baa-a285-ec850e0230f2">
